components/script/dom/htmlsourceelement.rs

```diff
@@ -46,6 +49,21 @@ impl VirtualMethods for HTMLSourceElement {
         Some(self.upcast::<HTMLElement>() as &VirtualMethods)
     }
 
+    fn attribute_mutated(&self, attr: &Attr, mutation: AttributeMutation) {
+        self.super_type().unwrap().attribute_mutated(attr, mutation);
+        match attr.local_name() {
+            &local_name!("srcset") | &local_name!("sizes")  |
+            &local_name!("media") | &local_name!("type") => {
+                if let Some(next_sibling) = self.upcast::<Node>().GetNextSibling() {
+                    if let Some(htmlimageelement_sibling) = next_sibling.downcast::<HTMLImageElement>() {
+                        htmlimageelement_sibling.update_the_image_data();
+                    }
+                }
+            },
+            _ => {},
+        }
+    }
+
     /// <https://html.spec.whatwg.org/multipage/#the-source-element:nodes-are-inserted>
     fn bind_to_tree(&self, tree_in_doc: bool) {
```

Comment:
We will need to implement the "The element's parent is a picture element and a source element is inserted as a previous sibling" mutation check in bind_to_truee and "The element's parent is a picture element and a source element that was a previous sibling is removed" in unbind_from_tree.

Reply:
Yup, got it!